v2/yarpcthrift/thriftrw-plugin-yarpcv2/fx_test.go

```diff
@@ -20,173 +20,176 @@
-// 	v, ok := h[*k]
-// 	if !ok {
-// 		return 0, &atomic.KeyDoesNotExist{Key: k}
-// 	}
-// 	return v, nil
-// }
+import (
+	"context"
+	"net"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxtest"
+	"go.uber.org/thriftrw/ptr"
+	"go.uber.org/yarpc/v2"
+	"go.uber.org/yarpc/v2/yarpcclient"
+	"go.uber.org/yarpc/v2/yarpchttp"
+	"go.uber.org/yarpc/v2/yarpcjson"
+	"go.uber.org/yarpc/v2/yarpcrouter"
+	"go.uber.org/yarpc/v2/yarpcthrift/thriftrw-plugin-yarpcv2/internal/tests/atomic"
+	"go.uber.org/yarpc/v2/yarpcthrift/thriftrw-plugin-yarpcv2/internal/tests/atomic/readonlystorefx"
+	"go.uber.org/yarpc/v2/yarpcthrift/thriftrw-plugin-yarpcv2/internal/tests/atomic/readonlystoreserver"
+	"go.uber.org/yarpc/v2/yarpcthrift/thriftrw-plugin-yarpcv2/internal/tests/atomic/storeclient"
+	"go.uber.org/yarpc/v2/yarpcthrift/thriftrw-plugin-yarpcv2/internal/tests/atomic/storefx"
+)
+
+func TestFxClient(t *testing.T) {
+	d := yarpcclient.NewProvider()
+	d.Register("store", yarpc.Client{})
+
+	assert.NotPanics(t, func() {
+		p := storefx.Params{
+			Provider: d,
+		}
+		f := storefx.Client("store").(func(storefx.Params) (storefx.Result, error))
+		f(p)
+	}, "failed to build client")
+
+	assert.Panics(t, func() {
+		f := storefx.Client("not-store").(func(yarpc.ClientProvider) storeclient.Interface)
+		f(d)
+	}, "expected panic")
+}
+
+func extractProcedures(procs *[]yarpc.Procedure) fx.Option {
+	type params struct {
+		fx.In
+
+		// We need to handle both cases: A single transport.Procedure provided
+		// to the "yarpcfx" group and a []transport.Procedure provided to the
+		// "yarpcfx" group.
+		SingleProcedures []yarpc.Procedure   `group:"yarpcfx"`
+		ProcedureLists   [][]yarpc.Procedure `group:"yarpcfx"`
+	}
+
+	return fx.Invoke(func(p params) {
+		for _, proc := range p.SingleProcedures {
+			*procs = append(*procs, proc)
+		}
+		for _, procList := range p.ProcedureLists {
+			*procs = append(*procs, procList...)
+		}
+	})
+}
+
+func echoJSON(ctx context.Context, req map[string]interface{}) (map[string]interface{}, error) {
+	return req, nil
+}
+
+func TestFxServer(t *testing.T) {
+	type jsonProcedures struct {
+		fx.Out
+
+		Procedures []yarpc.Procedure `group:"yarpcfx"`
+	}
+
+	handler := readOnlyStoreHandler{
+		"foo":    1,
+		"bar":    2,
+		"answer": 42,
+	}
+
+	var procedures []yarpc.Procedure
+	serverApp := fxtest.New(t,
+		fx.Provide(
+			func() readonlystoreserver.Interface { return handler },
+			readonlystorefx.Server(),
+			func() jsonProcedures {
+				return jsonProcedures{Procedures: yarpcjson.Procedure("echoJSON", echoJSON)}
+			},
+		),
+		extractProcedures(&procedures),
+	)
+	defer serverApp.RequireStart().RequireStop()
+
+	router := yarpcrouter.NewMapRouter("myserver")
+	router.Register(procedures)
+	listener, err := net.Listen("tcp", ":0")
+	require.NoError(t, err)
+	serverD := &yarpchttp.Inbound{
```

Comment:
nit: D is in reference to the deprecated `yarpc.Dispatcher` object.

Let's just call this `httpInbound` or `inbound`